akka-sample-fsm-java/src/main/java/sample/Chopstick.java

```diff
@@ -0,0 +1,125 @@
+package sample;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+
+public class Chopstick {
+    interface ChopstickMessage { }
+
+    final static class Take implements ChopstickMessage {
+        public final ActorRef<ChopstickAnswer> hakker;
+
+        public Take(ActorRef<ChopstickAnswer> hakker) {
+            this.hakker = hakker;
+        }
+    }
+
+    final static class Put implements ChopstickMessage {
+        public final ActorRef<ChopstickAnswer> hakker;
+
+        public Put(ActorRef<ChopstickAnswer> hakker) {
+            this.hakker = hakker;
+        }
+    }
+
+    public static Behavior<ChopstickMessage> create() {
+        return Behaviors.setup(context-> new Chopstick(context).available());
+    }
+
+    interface ChopstickAnswer {
```

Comment:
same here

Reply:
Looks like an abstract class with a chopstick field could be less verbose?